relayer/chain/substrate/listener.go

```diff
@@ -103,37 +94,39 @@ func (li *Listener) pollBlocks(ctx context.Context) error {
 				continue
 			}
 
-			// Get hash for latest block, sleep and retry if not ready
-			hash, err := li.conn.api.RPC.Chain.GetBlockHash(currentBlock)
+			digestItem, err := digest.GetAuxiliaryDigestItem(finalizedHeader.Digest)
 			if err != nil {
-				li.log.WithFields(logrus.Fields{
-					"error": err,
-					"block": currentBlock,
-				}).Error("Failed to fetch block hash")
-				sleep(ctx, retryInterval)
-				continue
-			}
-
-			var records types.EventRecordsRaw
-			_, err = li.conn.api.RPC.State.GetStorage(storageKey, &records, hash)
-			if err != nil {
-				li.log.WithError(err).Error("Failed to fetch events for block")
-				sleep(ctx, retryInterval)
-				continue
+				return err
 			}
 
-			li.log.WithField("record", hex.EncodeToString(records)).Trace("Fetched event record")
-
-			events, err := li.eventDecoder.Decode(records)
-			if err != nil {
+			if digestItem != nil && digestItem.IsCommitmentHash {
 				li.log.WithFields(logrus.Fields{
-					"error": err,
-					"block": currentBlock,
-				}).Error("Failed to decode events for block")
-				return err
-			}
+					"block":          finalizedHeader.Number,
+					"commitmentHash": digestItem.AsCommitmentHash.Hex(),
+				}).Debug("Found commitment hash in header digest")
+
+				storageKey, err := offchain.MakeStorageKey(digestItem.AsCommitmentHash)
+				if err != nil {
+					return err
+				}
+
+				data, err := li.conn.api.RPC.Offchain.LocalStorageGet(rpcOffchain.Persistent, storageKey)
+				if err != nil {
+					li.log.WithError(err).Error("Failed to read commitment from offchain storage")
+					sleep(ctx, retryInterval)
+					continue
+				}
+
+				if data != nil {
+					li.log.WithFields(logrus.Fields{
+						"block":               finalizedHeader.Number,
+						"commitmentSizeBytes": len(*data),
+					}).Debug("Retrieved commitment from offchain storage")
+				} else {
+					li.log.WithError(err).Error("Commitment not found in offchain storage")
+				}
 
-			li.handleEvents(currentBlock, events)
```

Comment:
will need li.handleCommitment(...), though can do in follow up PR. see https://github.com/Snowfork/polkadot-ethereum/pull/219/files#diff-42f6d75aa7d06e027e7684545996c8f782c82c6fc53f288c6c5852c6e26bfb71R36 for the expected interface to submit it to solidity